gradle/gooeycli/src/main/groovy/org/terasology/cli/commands/module/GetCommand.groovy

```diff
@@ -3,92 +3,77 @@
 
 package org.terasology.cli.commands.module
 
-import groovy.io.FileType
-import org.eclipse.jgit.api.Git
-import org.terasology.cli.ModuleItem
-import org.terasology.cli.options.GitOptions
-import org.terasology.cli.util.Constants
-import org.terasology.cli.util.ModuleIndex
+import org.terasology.cli.commands.items.ItemCommand
+import org.terasology.cli.items.ModuleItem
+import org.terasology.cli.module.ModuleIndex
+import org.terasology.cli.module.Modules
 import picocli.CommandLine
 import picocli.CommandLine.Command
-import picocli.CommandLine.Mixin
 import picocli.CommandLine.Option
 import picocli.CommandLine.Parameters
 
 @Command(name = "get", description = "get module ")
 class GetCommand implements Runnable {
```

Comment:
So if I'm looking at things right there are two `GetCommand` classes, one common one module, and the one selected is simply based on the import statement? I guess that works but it looks scary to me :-)

If dependencies is the only unique way they differ, would it make sense to abstract out a `ProvidesDependencies` interface or something with a method, and if during the get command setup we see that only then look at the dependency stuff? Then you could in theory support dependencies per type arbitrarily

Reply:
wonder what you gain by making this so abstract. trying to add additional layers to best fit the situation. wonder if it would just make more sense to implement this 3 or so times with some basic containers. feels a bit too clever for its own good.